### PR TITLE
Add context broker executable management (local and remote)

### PR DESCRIPTION
**Pull request by jcalderin**
Add Background to all features to start the CB before the tests
Add properties to set the context broker bin configuration

### CODE REVIEW

File: test/acceptance/integration/ISSUE-322 Service support for registries (NGSI9)/issue-322.feature [hunk 1]
```diff
@@ -350,8 +347,7 @@ Feature: Service support for registries (NGSI9)
     And the path in the last mock petition contains "service2"
     And clean the mongo database of the service "issue_322"
     And clean the mongo database of the service "issue_322_2"
-
-  @issue-322
+  @act
```
Comment (fgalan) [inline test/acceptance/integration/ISSUE-322 Service support for registries (NGSI9)/issue-322.feature]: Curiosity... what does this `@act` mean?

Comment (jcalderin) [inline test/acceptance/integration/ISSUE-322 Service support for registries (NGSI9)/issue-322.feature]: Ups, its my tag for debug.
Fixed in 741aa0e

File: test/acceptance/integration/tools/general_utils.py [hunk 2]
```diff
@@ -153,3 +154,160 @@ def drop_all_test_databases(ip, port):
...
+    set_ssh_config(localhost)
+    # Check if there is CB running
+    if localhost:
+        if world.cb_pid != '':
+            if local('ps -p {pid} | grep contextBroker'.format(pid=world.cb_pid)) != '':
+                local('kill -15 {pid} && sleep 5'.format(pid=world.cb_pid))
+                if local('ps -p {pid} | grep contextBroker'.format(pid=world.cb_pid)) != '':
+                    local('kill -9 {pid} && sleep 5'.format(pid=world.cb_pid))
+                    if local('ps -p {pid} | grep contextBroker'.format(pid=world.cb_pid)) != '':
+                        raise ValueError('After try to kill pep, is still running ')
```
Comment (fgalan) [inline test/acceptance/integration/tools/general_utils.py]: Is this message correct? It talks about pep, not contextBroker...

File: test/acceptance/integration/tools/general_utils.py [hunk 3]
```diff
@@ -153,3 +154,160 @@ def drop_all_test_databases(ip, port):
...
+    else:
+        if world.cb_pid != '':
+            if run('ps -p {pid} | grep contextBroker'.format(pid=world.cb_pid)) != '':
+                run('kill -15 {pid} && sleep 5'.format(pid=world.cb_pid))
+                if run('ps -p {pid} | grep contextBroker'.format(pid=world.cb_pid)) != '':
+                    run('kill -9 {pid} && sleep 5'.format(pid=world.cb_pid))
+                    if run('ps -p {pid} | grep contextBroker'.format(pid=world.cb_pid)) != '':
+                        raise ValueError('After try to kill pep, is still running ')
```
Comment (fgalan) [inline test/acceptance/integration/tools/general_utils.py]: Is this message correct? It talks about pep, not contextBroker...

File: test/acceptance/integration/tools/general_utils.py [hunk 4]
```diff
@@ -153,3 +154,160 @@ def drop_all_test_databases(ip, port):
...
+    :return:
+    """
+    config = world.config['deploy_data']
+    if not config['host'] == 'localhost' or not config['host'] == '127.0.0.1':
+        localhost = False
+    else:
+        localhost = True
+    set_ssh_config(localhost)
+    # Check if there is CB running
+    if localhost:
```
Comment (fgalan) [inline test/acceptance/integration/tools/general_utils.py]: Regarding stopping strategy... CB stores its pid in a file (check `-pidpath` CLI argument). Maybe you should start checking that before trying the ps/grep way.

File: test/acceptance/integration/tools/general_utils.py (hunk 3 above)
Comment (jcalderin) [inline test/acceptance/integration/tools/general_utils.py]: Fidex in 06a9635

File: test/acceptance/integration/tools/general_utils.py (hunk 2 above)
Comment (jcalderin) [inline test/acceptance/integration/tools/general_utils.py]: Fidex in 06a9635

File: test/acceptance/integration/tools/general_utils.py (hunk 4 above)
Comment (jcalderin) [inline test/acceptance/integration/tools/general_utils.py]: Fixed in 5670c19

**Comment by fgalan** [comment]
LGTM (passing the ball to @iariasleon for final LGTM + merge + delete branch)

**Comment by iariasleon** [comment]
LGTM

**Pull request merged**